loopy/transform/callable.py

```diff
@@ -595,9 +595,14 @@ def _shape_1_if_empty(shape):
         new_callee_insns = []
         for callee_insn in callee_knl.instructions:
             if isinstance(callee_insn, MultiAssignmentBase):
-                new_callee_insns.append(callee_insn.copy(expression=dim_changer(
-                    callee_insn.expression),
-                    assignee=dim_changer(callee_insn.assignee)))
+                if isinstance(callee_insn, CallInstruction):
+                    new_callee_insns.append(callee_insn.copy(expression=dim_changer(
+                        callee_insn.expression),
+                        assignees=dim_changer(callee_insn.assignees)))
+                else:
+                    new_callee_insns.append(callee_insn.copy(expression=dim_changer(
+                        callee_insn.expression),
+                        assignee=dim_changer(callee_insn.assignee)))
```

Comment:
How are `insn.assignee` and `insn.assignees` different? AKA: should we just use `insn.assignees` in both cases?

Reply:
Because https://github.com/inducer/loopy/blob/a42a1e6675670f60fec274716f938646ebf8f39f/loopy/kernel/instruction.py#L900 and https://github.com/inducer/loopy/blob/6832acfbe666f66851971754672c3b6a3e8a73b0/loopy/kernel/instruction.py#L1060 and we are special casing on the parent class of CallInstruction and Assignment.

Reply:
Actually we should probably drop the `if isinstance(callee_insn, MultiAssignmentBase):` I think it can only be either a CallInstruction or an Assignment

Reply:
@sv2518: How about:
```python
new_callee_insns.append(callee_insn
                        .with_transformed_expressions(dim_changer))
```
without any type-checking on the instruction type.

Reply:
> How are `insn.assignee` and `insn.assignees` different? AKA: should we just use `insn.assignees` in both cases?

`assignees` is the more general property that you can retrieve from anything that's a `MultiAssignmentBase`.

Reply:
I'm in favor of Kaushik's change here.

Reply:
Ok, done that now.